src/server/debug/server/server.go

```diff
@@ -927,3 +955,150 @@ func redirectDump(ctx context.Context, c debug.DebugClient, filter *debug.Filter
 	}
 	return grpcutil.NewStreamingBytesReader(dumpC, nil), nil
 }
+
+func (s *debugServer) collectDatabaseStats(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	if err := s.collectDatabaseActivities(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseRowCount(ctx, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseTableSizes(ctx, tw, prefix...); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *debugServer) collectDatabaseActivities(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "activities", queryDatabaseActivities, prefix...)
+}
+
+func (s *debugServer) collectDatabaseRowCount(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "row-counts", queryDatabaseRowCounts, prefix...)
+}
+
+func (s *debugServer) collectDatabaseTableSizes(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	return s.collectDatabaseQuery(ctx, tw, "table-sizes", queryDatabaseTableSizes, prefix...)
+}
+
+func (s *debugServer) collectDatabaseQuery(ctx context.Context, tw *tar.Writer, fileName string, queryFunc func(conn *sql.Conn, w io.Writer) error, prefix ...string) error {
+	return collectDebugFile(tw, fileName, "txt", func(w io.Writer) error {
+		conn, err := s.env.GetDBClient().Conn(ctx)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		defer conn.Close()
+
+		err = queryFunc(conn, w)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		return errors.EnsureStack(err)
+	}, prefix...)
+}
+
+func queryDatabaseRowCounts(conn *sql.Conn, w io.Writer) error {
+	query := `
+		SELECT schemaname,relname,n_live_tup 
+		  FROM pg_stat_user_tables 
+		ORDER BY n_live_tup DESC;
+	`
+
+	return queryDatabase(conn, query, w)
+}
+
+func queryDatabaseActivities(conn *sql.Conn, w io.Writer) error {
+	query := `
+		SELECT current_timestamp - query_start as runtime, datname, usename
+		FROM pg_stat_activity
+		WHERE state != 'idle'
+		ORDER by runtime desc
+		LIMIT 100;
+	`
+
+	return queryDatabase(conn, query, w)
+}
+
+func queryDatabaseTableSizes(conn *sql.Conn, w io.Writer) error {
+	query := `
+		SELECT nspname || '.' || relname AS "relation",
+		pg_size_pretty(pg_total_relation_size(C.oid)) AS "total_size"
+		FROM pg_class C
+		LEFT JOIN pg_namespace N ON (N.oid = C.relnamespace)
+		WHERE nspname NOT IN ('pg_catalog', 'information_schema')
+		AND C.relkind <> 'i'
+		AND nspname !~ '^pg_toast'
+		ORDER BY pg_total_relation_size(C.oid) DESC
+		LIMIT 5;
+	`
+
+	return queryDatabase(conn, query, w)
+}
+
+func queryDatabase(conn *sql.Conn, query string, w io.Writer) error {
+	rows, err := conn.QueryContext(context.Background(), query)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+
+	return writeRows(rows, w)
+}
+
+func writeRows(rows *sql.Rows, w io.Writer) error {
+	cols, err := rows.Columns()
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+
```

Comment:
nit: I think we generally want to avoid adding extra newlines in functions, so I'd remove these. That being said we haven't been explicit about it.

Reply:
Ahh gotcha, thanks for the style stuff. I will have to definitely deprogram some old habits 😆